sql/core/src/test/resources/sql-tests/inputs/non-excludable-rule.sql

```diff
@@ -4,3 +4,12 @@ SELECT
   (SELECT min(id) FROM range(10)),
   (SELECT sum(id) FROM range(10)),
   (SELECT count(distinct id) FROM range(10));
+
+-- SPARK-39444
+SET spark.sql.optimizer.excludedRules=org.apache.spark.sql.catalyst.optimizer.Optimizer$OptimizeSubqueries;
```

Comment:
Although it was working, I didn't realize that we allow to exclude rules whose prefix is `Optimizer$...`. Is it legitimate? I'm just curious if we need to ban all exclusions for ``Optimizer$...` rules.

Reply:
Another rule is: `org.apache.spark.sql.catalyst.optimizer.Optimizer$UpdateCTERelationStats`.